keyring/cli.py

```diff
@@ -86,10 +86,28 @@ def do_get(self):
         print(password)
 
     def do_set(self):
-        password = self.input_password(
-            f"Password for '{self.username}' in '{self.service}': "
-        )
-        set_password(self.service, self.username, password)
+        trials = 3
+        for i in range(trials):
+            password = self.input_password(
+                f"password for '{self.username}' in '{self.service}': "
+            )
+
+            reentered_password = self.input_password(
```

Comment:
I've started work on tests and run into an issue. The `input_password` method is coded to accept the input from a pipe (if present). This change of expecting two passwords when it's piped in is unacceptable. Probably instead we'll want to hook this logic in at `getpass.getpass` instead of wrapping `input_password`.

Reply:
In #663, I'm working on adding some tests to capture at least the current expectation and hopefully serve as a foundation for testing this more sophisticated behavior. Interestingly, those tests didn't seem to capture the concern I mentioned above, but I expected them to, so I'll need to think about it a bit more before proceeding.

I've also worked on some refactorings in the `feature/password-verification` branch. I'd pushed them to this PR, but decided to leave them in their own branch for now while working out this issue.

Reply:
Sure. Please let me know if you need help with any other issues as well. I'll be more than happy to contribute.

Reply:
Maybe we should ask to re-enter password only if `sys.stdin.isatty()`?

Reply:
Yeah I think that works. I think it would be a good idea to skip verification for if not typed in explicitly. I can make this change if it's fine with @jaraco. I would imagine the piped password would already be visual to the user in some form so the verification would not make much sense in that case.

Reply:
Hi @jaraco. Any update on this?

Reply:
Sorry for the delay. Yes, please proceed with drafting an implementation that avoids the re-prompt for non-interactive cases.